src/carousel/carousel.js

```diff
@@ -421,11 +420,24 @@ angular.module('ui.bootstrap.carousel')
 
 .value('$carouselSuppressWarning', false)
 
+.controller('CarouselController', ['$scope', '$element', '$controller', '$log', '$carouselSuppressWarning', function($scope, $element, $controller, $log, $carouselSuppressWarning) {
+  if (!$carouselSuppressWarning) {
+    $log.warn('CarouselController is now deprecated. Use UibCarouselController instead.');
+  }
+
+  angular.extend(this, $controller('UibCarouselController', {
```

Comment:
Instead of extending `this`, you can probably return the `$controller('Uib…` directly.